src/util/ConfigManager/ConfigManager.cpp

```diff
@@ -34,24 +34,70 @@
 #include "util/json.h"
 
 namespace ad_utility {
-static auto configurationOptionsImpl(auto& configurationOptions) {
-  return std::views::transform(configurationOptions, [](auto& pair) {
-    // Make sure, that there is no null pointer.
-    AD_CORRECTNESS_CHECK(pair.second);
+// ____________________________________________________________________________
+template <typename ReturnPointer>
+std::vector<std::pair<std::string, ReturnPointer>>
+ConfigManager::configurationOptionsImpl(auto& configurationOptions,
+                                        std::string_view pathPrefix) {
+  std::vector<std::pair<std::string, ReturnPointer>> collectedOptions;
 
-    // Return a dereferenced reference.
-    return std::tie(pair.first, *pair.second);
-  });
+  std::ranges::for_each(
+      configurationOptions,
+      [&collectedOptions, &pathPrefix](auto pair) {
+        const std::string& pathToCurrentEntry =
+            absl::StrCat(pathPrefix, std::get<0>(pair));
+
+        std::visit(
+            [&collectedOptions, &pathToCurrentEntry]<typename T>(T& var) {
+              // A normal `ConfigOption` can be directly added. For a
+              // `ConfigManager` we have to recursively collect the options.
+              if constexpr (std::is_same_v<std::decay_t<T>, ConfigOption>) {
+                collectedOptions.push_back(
+                    std::make_pair(pathToCurrentEntry, &var));
+              } else {
+                AD_CORRECTNESS_CHECK(
+                    (std::is_same_v<std::decay_t<T>, ConfigManager>));
```

Comment:
Can be `static_assert` and `isSimilar` (see above).

Reply:
Done, see [e01b09b](https://github.com/ad-freiburg/qlever/pull/1071/commits/e01b09b9d9088b196780e130a8f99401d7da8483).